### PR TITLE
feature: implement StopContainer, RemoveContainer and ListContainers of cri manager

### PR DESCRIPTION
**Pull request by YaoZengzeng**
Signed-off-by: YaoZengzeng <yaozengzeng@zju.edu.cn>

**1.Describe what this PR did**

**2.Does this pull request fix one issue?** 

fixes part of #420 

**3.Describe how you did it**

**4.Describe how to verify it**

Maybe the [cri-tools](https://github.com/kubernetes-incubator/cri-tools) is what you need.

We may use the following method to verify that we can stop, remove and list container in sandbox:
```
// Now we must ensure the sandbox image has already exists.
[root@pouch-test cri-tools]# pouch pull k8s.gcr.io/pause-amd64:3.0
[root@pouch-test cri-tools]# cat sandbox-config.json
{
    "metadata": {
        "name": "nginx-sandbox",
        "namespace": "default",
        "attempt": 1,
        "uid": "hdishd83djaidwnduwk28bcsb"
    },
    "linux": {
    }
}
[root@pouch-test cri-tools]# crictl runs sandbox-config.json
008cc69c9c9c8e9c4d78aa8d6c528e21ab26affdc64748f631fb5dcb59963216
[root@pouch-test cri-tools]# cat container-config.json 
{
  "metadata": {
      "name": "busybox"
  },
  "image":{
      "image": "docker.io/library/busybox:latest"
  },
  "command": [
      "top"
  ],
  "linux": {
  }
}
[root@pouch-test cri-tools]# crictl create 008cc69c9c9c8e9c4d78aa8d6c528e21ab26affdc64748f631fb5dcb59963216 container-config.json sandbox-config.json
3530bbb414ccce09ba52fbdf781df53b881661720f05d92edf8820280abf064d
[root@pouch-test cri-tools]# crictl start 3530bbb414ccce09ba52fbdf781df53b881661720f05d92edf8820280abf064d
[root@pouch-test cri-tools]# crictl ps
CONTAINER ID        IMAGE                              CREATED             STATE               NAME                ATTEMPT
270b4d34783fb       docker.io/library/busybox:latest   48 years ago        CONTAINER_RUNNING   busybox             0
[root@pouch-test cri-tools]# crictl stop 3530bbb414ccc
[root@pouch-test cri-tools]# pouch ps
Name                                                            ID       Status    Image                              Runtime   Created
k8s_busybox_nginx-sandbox_default_hdishd83djaidwnduwk28bcsb_0  3530bb   stopped   docker.io/library/busybox:latest   runc      34 seconds ago
k8s_POD_nginx-sandbox_default_hdishd83djaidwnduwk28bcsb_1       30a21b   running   k8s.gcr.io/pause-amd64:3.0         runc      1 minute ago
[root@pouch-test cri-tools]# crictl rm 3530bbb414ccc
[root@pouch-test cri-tools]# pouch ps
Name                                                        ID       Status    Image                        Runtime   Created
k8s_POD_nginx-sandbox_default_hdishd83djaidwnduwk28bcsb_1   30a21b   running   k8s.gcr.io/pause-amd64:3.0   runc      1 minute ago
```

**5.Special notes for reviews**

### CODE REVIEW

**Comment by YaoZengzeng** [comment]
@allencloud  PTAL.

File: daemon/mgr/cri.go [hunk 1]
```diff
@@ -238,6 +278,26 @@ func makeContainerName(s *runtime.PodSandboxConfig, c *runtime.ContainerConfig)
 	}, nameDelimiter)
 }
 
+func parseContainerName(name string) (*runtime.ContainerMetadata, error) {
+	parts := strings.Split(name, nameDelimiter)
+	if len(parts) != 6 {
+		return nil, fmt.Errorf("failed to parse container name: %q", name)
```
Comment (allencloud) [inline daemon/mgr/cri.go]: I think here we only throw the error, but have not told users the root cause of the error.

How about ` fmt.Errorf("failed to parse container name %q: name format must be in format blablabla", name)`

**Comment by allencloud** [comment]
CI fails @YaoZengzeng

**Comment by YaoZengzeng** [comment]
@allencloud updated.

Maybe we should move tool functions like `extractLabels`, `parseUint32`... to another file like `cri_utils.go` and we'll do the unit test in file like `cri_test.go`, which would make file `cri.go` more clean.

I'll do it in another PR, is this OK?

**Comment by allencloud** [comment]
> Maybe we should move tool functions like extractLabels, parseUint32... to another file like cri_utils.go

Totally agree. I think it looks not so clear in dir mgr currently.

**Comment by allencloud** [comment]
LGTM, please add a PR for unit test ASAP. Thanks.

**Pull request merged**